CRM/Mailing/Event/BAO/Opened.php

```diff
@@ -241,11 +241,24 @@ public static function &getRows(
     $contact = CRM_Contact_BAO_Contact::getTableName();
     $email = CRM_Core_BAO_Email::getTableName();
 
+    $selectClauses = array(
+      "$contact.display_name as display_name",
+      "$contact.id as contact_id",
+      "$email.email as email",
+      "{$open}.time_stamp as date",
+    );
+
+    if ($is_distinct) {
+      $groupBy = " GROUP BY $queue.id ";
+      $select = CRM_Contact_BAO_Query::appendAnyValueToSelect($selectClauses, "{$open}.time_stamp as date");
```

Comment:
For this purpose, I think min(time_stamp) is preferred as we would want the earliest open event over the latest (i.e. the first time the user opened the email is the first "unique" event).